src/Aop/Framework/AbstractMethodInvocation.php

```diff
@@ -93,6 +93,11 @@ public function __construct(string $className, string $methodName, array $advice
             if ($this->level > 0) {
                 [$this->arguments, $this->instance, $this->current] = array_pop($this->stackFrames);
             }
+            else {
+                $this->arguments = [];
+                $this->instance = null;
+                $this->current = null;
```

Comment:
Please, align all consecutive assignments. Also, `$this->current `is updated each time in the recursive calls, I'm not sure that your patch support recursive calls. I would suggest to drop this line entirely, clear only arguments and instance variables